nano/core_test/node.cpp

```diff
@@ -3317,7 +3317,9 @@ TEST (node, block_processor_signatures)
 				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
 				 .work (*node1.work_generate_blocking (send3->hash ()))
 				 .build_shared ();
-	send5->signature.bytes[31] ^= 0x1;
+	auto signature = send5->block_signature ();
+	signature.bytes[31] ^= 0x1;
+	send5->signature_set (signature);
```

Comment:
I agree it's more clean to get the signature, touch it, then set it back, but was there a functional problem with changing it in-place?

Reply:
Since we're holding a pointer to nano::block instead of the specific block type, the signature field is no longer directly accessible.